packages/icons/src/Icon.tsx

```diff
@@ -0,0 +1,73 @@
+import { useResponsiveValue, useStyledSystem, type ResponsiveProp, type StyledComponentProps } from "@hopper-ui/styled-system";
+import { filterDOMProps } from "@react-aria/utils";
+import clsx from "clsx";
+import { createContext, forwardRef, type ElementType, type RefAttributes, type SVGProps } from "react";
+import { useContextProps, type ContextValue, type SlotProps } from "react-aria-components";
+import styles from "./Icon.module.css";
+
+export interface IconProps extends SlotProps, StyledComponentProps<"svg"> {
+    /**
+    * The size of the icon.
+    */
+    size?: ResponsiveProp<"sm" | "md" | "lg">;
```

Comment:
ResponsiveProp is the name we use for props that allows different values based on the screen's width: https://wl-orbiter-website.netlify.app/?path=/docs/responsive-styles--page#usage

This allow `<AddIcon size={{base: "sm", "xl": "md"}} />` so the icon is sm in mobile, but md in 1440px+

Reply:
Are there any cases where this could happen?

Reply:
I feel like changing header sizes and icon size in mobile is one of the most frequent use cases. 

I also think that most of our props should allow responsive values when it can make sense (props like isOpen, isDisabled, onChange don't require to be responsive, but anything related to styling should support it)